lib/services/platform-service.ts

```diff
@@ -461,7 +461,9 @@ export class PlatformService implements IPlatformService {
 
 	public deployOnDevice(platform: string, buildConfig?: IBuildConfig): IFuture<void> {
 		return (() => {
-			this.installOnDevice(platform, buildConfig).wait();
+			if (!this.$options.debug) {
+				this.installOnDevice(platform, buildConfig).wait();
```

Comment:
Thanks for being honest. I excluded all the logic related to the debug service and moved it in the debug command. Now the single responsibility principle is not hurt. The debug command relies on messages to communicate with the livesync service. I checked your approach, it will not work when we use the --watch option.